### ISSUE TITLE
[sqlvm-alwayson-cluster] Loadbalancer creation failure

### ISSUE THREAD

**Issue by guysidford** [Azure/azure-quickstart-templates#5337]
### Issue Details

When deploying to a clean resource group in useast2, I am receiving the following failure when the LoadBalancer is being created.

`{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.","details":[{"code":"BadRequest","message":"{\r\n \"error\": {\r\n \"code\": \"InvalidResourceReference\",\r\n \"message\": \"Resource /subscriptions/xxxxxxxxxx/resourceGroups/RG_USEAST2/providers/Microsoft.Network/virtualNetworks/AUTOHAV2VNET referenced by resource /subscriptions/xxxxxxxx/resourceGroups/rg_useast2/providers/Microsoft.Network/loadBalancers/sqlLoadBalancer was not found. `

I can see that what is happening is that we define the network name in azuredeploy.json

```
"virtualNetworkName": { 
"type": "string", 
"metadata": {
 "description": "Name of virtual network to be created" 
},
 "defaultValue": "autohav2VNET"
```

it gets created in vnet-new.json, referencing the name we have passed.

`"resources": [ { "name": "[parameters('virtualNetworkName')]",`

But we get a network with a name like autohav2VNETl5g

And when creating the LoadBalancer we are looking for the name we have defined in the script rather than the name that it is created with.

...../virtualNetworks/AUTOHAV2VNET referenced by resource 

Why are we seeing this behaviour?

**Comment by guysidford** [comment]
I can see that what is happening is that we define the network name in azuredeploy.json

"virtualNetworkName": {
      "type": "string",
      "metadata": {
        "description": "Name of virtual network to be created"
      },
      "defaultValue": "autohav2VNET"

it gets created in vnet-new.json, referencing the name we have passed.

  "resources": [
    {
      "name": "[parameters('virtualNetworkName')]",

But we get a network with a name like autohav2VNETl5g

And when creating the LoadBalancer we are looking for the name we have defined in the script rather than the nae that it is created with.

"message": "Resource /subscriptions/b7dc1571-7092-4256-bd19-f04e2ad7ce77/resourceGroups/RG_USEAST2/providers/Microsoft.Network/virtualNetworks/AUTOHAV2VNET referenced by resource /subscriptions/b7dc1571-7092-4256-bd19-f04e2ad7ce77/resourceGroups/rg_useast2/providers/Microsoft.Network/loadBalancers/sqlLoadBalance

Why are we seeing this behaviour?
